quinn-udp/src/windows.rs

```diff
@@ -1,23 +1,46 @@
 use std::{
     io::{self, IoSliceMut},
     mem,
+    net::{IpAddr, Ipv4Addr},
     os::windows::io::AsRawSocket,
+    ptr,
     sync::Mutex,
     time::Instant,
 };
 
+use libc::{c_int, c_uint};
+use once_cell::sync::Lazy;
 use windows_sys::Win32::Networking::WinSock;
 
-use super::{log_sendmsg_error, RecvMeta, Transmit, UdpSockRef, IO_ERROR_LOG_INTERVAL};
+use crate::{
+    cmsg::{self, CMsgHdr},
+    log_sendmsg_error, EcnCodepoint, RecvMeta, Transmit, UdpSockRef, IO_ERROR_LOG_INTERVAL,
+};
 
-/// QUIC-friendly UDP interface for Windows
+/// Helper to perform advanced operations on a UDP socket
 #[derive(Debug)]
 pub struct UdpSocketState {
     last_send_error: Mutex<Instant>,
 }
 
 impl UdpSocketState {
+    /// Create a helper for `sock`
+    ///
+    /// # Panics
+    /// - If `CMSG_LEN` is too small to store a [`WinSock::IN6_PKTINFO`], one `libc::c_int` (for ECN)
+    ///   and one `u32` (for Segment size)
```

Comment:
On review, I note this is *also* concerned with implementation details that don't mean anything to users. Although this contradicts @djc's guidance, I think this note risks adding more confusion than it avoids: a reader should immediately ask what `CMSG_LEN` is and what they have to do about it.

Reply:
Ok, lets wait for @djc

Reply:
Maybe we could use a more generic documentation like for examples : 
# Panics
- If some operating system dependent invariant are not valid

Reply:
I dunno, that doesn't seem actionable for users. A panic here is always a bug on our part, IMO, and hence isn't really part of the public API, any more than a panic due to indexing out of bounds would be.

Reply:
We could replace these checks by [compile time errors](https://doc.rust-lang.org/std/macro.compile_error.html) ?

Reply:
That would be nice, but `libc`'s bindings of the cmsg macros aren't `const fn`s. In practice, test coverage on supported platforms is an adequate approximation of a compile-time error, IMO.

Reply:
So we could add this as two regression tests (one for windows and one for unix) instead of an assert within the `new()` ?

Reply:
I'm on board with reverting the addition of this note if this is not actionable for the caller.

Reply:
ok il will do it